### PR TITLE
feat(hogql): ignore future persons

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

https://posthog.slack.com/archives/C02E3BKC78F/p1716558850878309?thread_ts=1716503393.145799&cid=C02E3BKC78F

## Changes

Exclude persons with future created_at dates

## Does this work well for both Cloud and self-hosted?

I hope so.

## How did you test this code?

CI + snapshots.

Running `select * from persons` under `/debug` adds the following: `ifNull(less(person.created_at, plus(now64(6, %(hogql_val_6)s), toIntervalDay(1))), 0))`

### CODE REVIEW

**Review by tiina303 — APPROVED** [review]
Ship it, follow-up Q:
1. if folks do an insight query, then they there won't see the person show up in a modal either

File: posthog/hogql/database/argmax.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+from posthog.hogql.parser import parse_expr
+
 
 def argmax_select(
     table_name: str,
     select_fields: dict[str, list[str | int]],
     group_fields: list[str],
     argmax_field: str,
     deleted_field: Optional[str] = None,
+    no_future_field: Optional[str] = None,
```
Comment (Twixes) [inline posthog/hogql/database/argmax.py:13]: Would be useful to document the meaning of this in the function's docstring, as it's not immediately obvious, being very specific to the persons use case

**Review by Twixes — CHANGES_REQUESTED** [review]
Needs a test, but substance looks good

**Comment by mariusandra** [comment]
Any query anywhere selecting `persons` will now have this applied to the subquery we make. See [the generated SQL](https://us.posthog.com/project/2/debug#q=%7B%0A%20%20%22kind%22%3A%20%22HogQLQuery%22%2C%0A%20%20%22query%22%3A%20%22select%20*%20from%20persons%20limit%2010%22%2C%0A%20%20%22filters%22%3A%20%7B%0A%20%20%20%20%22dateRange%22%3A%20%7B%0A%20%20%20%20%20%20%22date_from%22%3A%20%22-24h%22%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) here.

image

File: posthog/hogql/database/argmax.py (hunk 1 above)
Comment (mariusandra) [reply posthog/hogql/database/argmax.py]: renamed to `timestamp_field_to_clamp` for now

**Review by Twixes — APPROVED** [review]

**Pull request merged**